src/main/kotlin/cacophony/grammars/Analysis.kt

```diff
@@ -2,43 +2,49 @@ package cacophony.grammars
 
 import cacophony.automata.DFA
 
-typealias DFAStateReference<StateType, AtomType, ResultType> = Pair<StateType, DFA<StateType, AtomType, ResultType>>
+typealias DFAStateReference<StateType, SymbolType, ResultType> =
+    Pair<StateType, DFA<StateType, SymbolType, ResultType>>
 
-fun <StateType, SymbolType> findNullable(
-    automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
-): Collection<DFAStateReference<StateType, SymbolType, Production<SymbolType>>> {
+typealias StateToSymbolsMap<StateType, SymbolType, ResultType> =
+    Map<DFAStateReference<StateType, SymbolType, ResultType>, Collection<SymbolType>>
+
+fun <StateType, SymbolType, ResultType> findNullable(
```

Comment:
`find....` methods should not care about the `ResultType` - the same sets of states should be returned.
It will be a bit easier to test if we can put any DFA without productions as results

Reply:
good point